dafny/lcf/datalog/AstBuilder.cs

```diff
@@ -70,16 +70,88 @@ internal class AstBuilder : datalogBaseVisitor<object>
       return context.Start.Line;
     }
 
-    public override object VisitClause(datalogParser.ClauseContext context) {
-      var name = Sequence<char>.FromString(context.name.Text);
+    public override object VisitBuiltin(datalogParser.BuiltinContext context) {
       var terms = (Dafny.ISequence<_ITerm>) VisitTerm_list(context.term_list());
-      return new _module.Prop_App(name, terms);
+      switch (context.name.Text) {
```

Comment:
All of these cases have the same form:

```
return new _module.Prop_BuiltinOp(new _module.Builtin_SplitString(), terms);
```

Can we refactor this to use function that returns the builtin type from the builtin name?

```
    switch (context.name.Text) {
        case "split_string":
          return new _module.Builtin_SplitString();
        case "sub_string":
          return new _module.Builtin_SubString();
        // ...
```

Then each case doesn't need to have the `new _module.Prop_BuiltinOp(..., terms);` repeated.

Reply:
Done